pom.xml

```diff
@@ -62,7 +62,7 @@
   </scm>
 
   <properties>
-    <jenkins.version>2.346.1</jenkins.version>
+    <jenkins.version>2.346.3</jenkins.version>
```

Comment:
Any reason for this specific upgrade?

Reply:
https://www.jenkins.io/doc/developer/plugin-development/choosing-jenkins-baseline/ but does not belong in this PR

Reply:
sure, about choosing, but for this PR I don't know what the plugin would benefit from the `2.346.3` that the `2.346.1` is not providing..

Reply:
@alecharp  required for the `workflow-*` test dependencies.